components/prism-factor.js

```diff
@@ -0,0 +1,303 @@
+(function (Prism) {
+
+	var factor = {
+
+		// ! single-line exclamation point comments with whitespace after/around the !
+		'comment': {
+			pattern: /(^|\s)(?:! .*|!$)/,
+			lookbehind: true,
+			inside: {
+				'bold': /\b(?:TODO|FIXME|NOTE|BUG|XXX|HACK)(?:\b|:)/
+			}
+		},
+
+		'number': [
+			{
+				// basic base 10 integers 9, -9
+				pattern: /(^|\s)[+-]?\d+(?=\s|$)/,
+				lookbehind: true
+			},
+			{
+				// base prefix integers 0b010 0o70 0xad 0d10 0XAD -0xa9
+				pattern: /(^|\s)[+-]?0(?:b[01]+|o[0-7]+|d\d+|x[\dA-F]+)(?=\s|$)/i,
+				lookbehind: true
+			},
+			{
+				// fractional ratios 1/5 -1/5 and the literal float approximations 1/5. -1/5.
+				pattern: /(^|\s)[+-]?\d+\/\d+\.?(?=\s|$)/,
+				lookbehind: true
+			},
+			{
+				// positive mixed numbers 23+1/5 +23+1/5
+				pattern: /(^|\s)\+?\d+\+\d+\/\d+(?=\s|$)/,
+				lookbehind: true
+			},
+			{
+				// negative mixed numbers -23-1/5
+				pattern: /(^|\s)-\d+-\d+\/\d+(?=\s|$)/,
+				lookbehind: true
+			},
+			{
+				// basic decimal floats -0.01 0. .0 .1 -.1 -1. -12.13 +12.13
+				// and scientific notation with base 10 exponents 3e4 3e-4 .3e-4
+				pattern: /(^|\s)[+-]?(?:\d*\.\d+|\d+\.\d*|\d+)(?:e[+-]?\d+)?(?=\s|$)/i,
+				lookbehind: true
+			},
+			{
+				// NAN literal syntax NAN: 80000deadbeef, NAN: a
+				pattern: /(^|\s)NAN:[\n\t ]+[\da-fA-F]+(?=\s|$)/,
+				lookbehind: true
+			},
+			{
+				/*
+					base prefix floats 0x1.0p3 (8.0) 0b1.010p2 (5.0) 0x1.p1 0b1.11111111111111111111111111p1111111111111111
+					"The normalized hex form ±0x1.MMMMMMMMMMMMM[pP]±EEEE allows any floating-point number to be specified precisely.
+					The values of MMMMMMMMMMMMM and EEEE map directly to the mantissa and exponent fields of the binary IEEE 754 representation."
+					<https://docs.factorcode.org/content/article-syntax-floats.html>
+				*/
+				pattern: /(^|\s)[+-]?0(?:b1\.[01]*|o1\.[0-7]*|d1\.\d*|x1\.[\dA-F]*)p\d+(?=\s|$)/i,
+				lookbehind: true
+			}
+		],
+
+		'boolean': {
+			pattern: /(^|\s)(?:t|f)(?=\s|$)/,
+			lookbehind: true
+		},
+
+		'special-string-literal': {
+			pattern: /(^|\s)[A-Z0-9\-]+"\s(?:\\"|\\\S|[^"\\])*"/,
```

Comment:
Please also apply the improvement to `string` here.
```suggestion
			pattern: /(^|\s)[A-Z0-9\-]+"\s(?:\\\S|[^"\\])*"/,
```

Reply:
Haha, I forgot about it because I committed in a hurry from my phone.